mobile/src/main/java/org/openhab/habdroid/ui/OpenHABPreferencesActivity.java

```diff
@@ -224,17 +243,86 @@ public boolean onPreferenceClick(Preference preference) {
                 }
             });
 
-            //fullscreen is not supoorted in builds < 4.4
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-                final PreferenceScreen ps = getPreferenceScreen();
-                ps.removePreference(ps.findPreference(Constants.PREFERENCE_FULLSCREEN));
+            ringtonePreference.setOnPreferenceChangeListener(new OnPreferenceChangeListener() {
+                @Override
+                public boolean onPreferenceChange(Preference preference, Object newValue) {
+                    updateRingtonePreferenceSummary(preference, newValue);
+                    return true;
+                }
+            });
+
+            final PreferenceScreen ps = getPreferenceScreen();
+
+            // Fullscreen is not supoorted in Android < 4.4
+            if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
+                Preference fullscreenPreference = ps.findPreference(Constants.PREFERENCE_FULLSCREEN);
+                fullscreenPreference.setEnabled(false);
+                fullscreenPreference.setSummary(R.string.settings_disabled_kitkat);
+            }
+
+            removeUnsupportedPrefs(ps);
```

Comment:
Why not move the remove code here and base it on CloudMessagingHelper.isSupported()?

Reply:
Initially I wanted to keep the "Not supported in foss" string in the foss strings file.

Reply:
Fair enough, but like mentioned in the other comment, I'd rather remove the preference completely. IMHO it's sufficient to inform the user about the existence of the full build in one place.